crates/hc_sandbox/tests/cli.rs

```diff
@@ -1,22 +1,57 @@
-use std::future::Future;
-use std::sync::Arc;
-
-// TODO: do the create tests first
-// TODO: run hc-sandbox then see if we can call the app websocket
-// TODO: Put holochain on the path
-// TODO:
-
 use assert_cmd::prelude::*;
 use holochain_conductor_api::AppRequest;
 use holochain_conductor_api::AppResponse;
 use holochain_websocket::{self as ws, WebsocketConfig, WebsocketReceiver, WebsocketSender};
 use matches::assert_matches;
+use once_cell::sync::Lazy;
 use portpicker::pick_unused_port;
+use std::future::Future;
+use std::path::PathBuf;
+use std::process::Stdio;
+use std::sync::Arc;
+use tokio::io::AsyncWriteExt;
 use tokio::process::Command;
 use url2::url2;
 
 const WEBSOCKET_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(3);
 
+static HC_BUILT_PATH: Lazy<PathBuf> = Lazy::new(|| {
+    let mut manifest_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    manifest_path.push("../hc/Cargo.toml");
+
+    let out = escargot::CargoBuild::new()
+        .bin("hc")
+        .current_target()
+        .current_release()
+        .manifest_path(manifest_path)
+        // Not defined on CI
+        .target_dir(PathBuf::from(
+            option_env!("CARGO_TARGET_DIR").unwrap_or("./target"),
+        ))
```

Comment:
@steveeJ - is there any better way to work around the nix building stuff? This is going to result in everything being rebuilt in a fresh target directory, right?

Reply:
Thanks to @steveeJ this is now in much better shape. If binaries are available the tests won't attempt to build them and if they are missing then it will build them once and use those instead. Note that this is once across test runs, as long as you don't change anything that would cause Cargo to do a new build